spec/integration/scope_builder_spec.rb

```diff
@@ -27,5 +27,31 @@
 
       expect(scope).to be_an_instance_of scope_class
     end
+
```

Comment:
:bulb: If it helps, in my local patch, I needed a way to test that the cache works as expected and ended up with the following spec:

<details>
  <summary>Spec Snippet</summary>
  
``` ruby
RSpec.describe Milestoner::Views::ScopeBuilder do
  subject(:builder) { described_class }

  let :view do
    Class.new Hanami::View do
      config.paths = Bundler.root.join "lib/milestoner/templates"
      config.template = "n/a"
    end
  end

  let(:scope_a) { Class.new Hanami::View::Scope }
  let(:scope_b) { Class.new Hanami::View::Scope }

  before do
    stub_const "TestScopeA", scope_a
    stub_const "TestScopeB", scope_b
    Hanami::View.cache.cache.clear
  end

  describe ".call" do
    it "caches scope without duplicates" do
      builder.call TestScopeA, locals: {}, rendering: view.new.rendering
      builder.call TestScopeB, locals: {}, rendering: view.new.rendering
      builder.call "TestScopeA", locals: {}, rendering: view.new.rendering
      builder.call "TestScopeB", locals: {}, rendering: view.new.rendering
      builder.call scope_a, locals: {}, rendering: view.new.rendering
      builder.call scope_b, locals: {}, rendering: view.new.rendering

      expect(Hanami::View.cache.cache.values).to eq([TestScopeA, TestScopeB])
    end
  end
end
```
</details>

Might be a good idea to add a spec for `#call` because that ensures the cache is used and doesn't duplicate itself. This also means you'd need to clear the cache between specs which why I put the in the `before` block.